api/managementpb/dbaas/psmdb_cluster.proto

```diff
@@ -0,0 +1,213 @@
+syntax = "proto3";
+
+package dbaas.v1beta1;
+
+option go_package = "api/managementpb/dbaas;dbaasv1beta1";
+
+import "github.com/mwitkow/go-proto-validators/validator.proto";
+import "google/api/annotations.proto";
+import "managementpb/dbaas/dbaas.proto";
+
+// PSMDBClusterState represents PSMDB cluster CR state.
+enum PSMDBClusterState {
+  // PSMDB_CLUSTER_STATE_INVALID represents unknown state.
+  PSMDB_CLUSTER_STATE_INVALID = 0;
+  // PSMDB_CLUSTER_STATE_CHANGING represents a cluster being changed (initializing).
+  PSMDB_CLUSTER_STATE_CHANGING = 1;
+  // PSMDB_CLUSTER_STATE_READY represents a cluster without pending changes (ready).
+  PSMDB_CLUSTER_STATE_READY = 2;
+  // PSMDB_CLUSTER_STATE_FAILED represents a failed cluster (error).
+  PSMDB_CLUSTER_STATE_FAILED = 3;
+  // PSMDB_CLUSTER_STATE_DELETING represents a cluster being deleting.
+  PSMDB_CLUSTER_STATE_DELETING = 4;
+}
+
+// PSMDBClusterParams represents PSMDB cluster parameters that can be updated.
+message PSMDBClusterParams {
+  // ReplicaSet container parameters.
+  message ReplicaSet {
+    // Requested compute resources.
+    ComputeResources compute_resources = 1;
+  }
+  // Cluster size.
+  int32 cluster_size = 1 [
+    (validator.field) = {
+      int_gt: 0
+    }
+  ];
+  // ReplicaSet container parameters.
+  ReplicaSet replicaset = 2 [
+    (validator.field) = {
+      msg_exists: true
+    }
+  ];
+}
+
+message ListPSMDBClustersRequest {
+  // Kubernetes cluster name.
+  string kubernetes_cluster_name = 1 [
+    (validator.field) = {
+      string_not_empty: true
+    }
+  ];
+}
+
+message ListPSMDBClustersResponse {
+  // Cluster represents PSMDB cluster information.
+  message Cluster {
+    // Cluster name.
+    string name = 1;
+    // Cluster state.
+    PSMDBClusterState state = 2;
+    // Currently running operation, if any.
+    RunningOperation operation = 3;
+    // Cluster parameters.
+    PSMDBClusterParams params = 4;
+  }
+  // PSMDB clusters information.
+  repeated Cluster clusters = 1;
+}
+
+message GetPSMDBClusterRequest {
+  // Kubernetes cluster name.
+  string kubernetes_cluster_name = 1 [
+    (validator.field) = {
+      string_not_empty: true
+    }
+  ];
+  // PSMDB cluster name.
+  string name = 2 [
+    (validator.field) = {
+      string_not_empty: true
+    }
+  ];
+}
+
+message GetPSMDBClusterResponse {
+  // PSMDBCredentials is a credentials to connect to PSMDB.
+  message PSMDBCredentials {
+    // MongoDB username.
+    string username = 1;
+    // MongoDB password.
+    string password = 2;
+    // MongoDB host.
+    string host = 3;
+    // MongoDB port.
+    int32 port = 4;
+    // Replicaset name.
+    string replicaset = 5;
```

Comment:
We call it `replication_set` in other APIs. Not sure if we should use the same name there